tempel.el

```diff
@@ -348,12 +363,30 @@ PROMPT is the optional prompt/default value."
 
 (defun tempel--save ()
   "Save template file buffer."
-  (when-let (buf (get-file-buffer tempel-file))
-    (with-current-buffer buf
-      (when (and (buffer-modified-p) (y-or-n-p (format "Save file %s? " tempel-file)))
-        (save-buffer buf)))))
-
-(defun tempel--file-read (file)
+  (let (modified)
+    (dolist (file (tempel--expand-path tempel-path))
+      (when-let (buffer (find-buffer-visiting file))
+    	(when (buffer-modified-p buffer)
+	  (push buffer modified))))
+    (cond ((not modified))
+	  ((not (cdr modified))
+	   (y-or-n-p "Save template file %s? " (car modified)))
+	  (t
+	   (y-or-n-p "Save modified template files? ")))
+    (mapc #'save-buffer modified)))
+
+(defun tempel--path-modified-p ()
```

Comment:
Instead of introducing this function `tempel--path-modified-p` my suggestion is to introduce a function `tempel--path-modification-time` which returns the maximum modification time of all the files in the path. Then you compare this time below.